### PR TITLE
Create book crud

### PR DESCRIPTION
**Pull request by juanwongramirez**
## Summary
- Add BookController
- Manage exceptions in finder methods

## Trello Card
https://trello.com/c/o7cWbkOc

### CODE REVIEW

File: src/main/java/wolox/training/exceptions/BookNotFoundException.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package wolox.training.exceptions;
+
+public class BookNotFoundException extends RuntimeException {
+
+    public BookNotFoundException(){
+        super("The book can not be found");
+    }
+
+    public BookNotFoundException(long id){
+        super(String.format("The book with id : %d, not found",id));
```
Comment (jeymarsonwolox) [inline src/main/java/wolox/training/exceptions/BookNotFoundException.java]: use `Constants.FORMAT_MESSSAGE_BOOK_NOT_FOUND` here

File: src/main/java/wolox/training/controllers/BookController.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    public Book findOne(@PathVariable Long id) {
+        return bookRepository.findById(id)
+                .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND,
+                        String.format(Constants.FORMAT_MESSSAGE_BOOK_NOT_FOUND,id)));
+    }
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public Book create(@RequestBody Book book) {
+        System.out.println(book.toString());
```
Comment (jeymarsonwolox) [inline src/main/java/wolox/training/controllers/BookController.java]: remove `sout` please

File: src/main/java/wolox/training/exceptions/BookIdMismatchException.java [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package wolox.training.exceptions;
+
+public class BookIdMismatchException extends RuntimeException{
+
+    public  BookIdMismatchException(){
+        super("The id value of book is not the same of the parameter");
```
Comment (jeymarsonwolox) [inline src/main/java/wolox/training/exceptions/BookIdMismatchException.java]: maybe you can extract this to a constant

**Review by jeymarsonwolox — DISMISSED** [review]

**Review by jeymarsonwolox — CHANGES_REQUESTED** [review]
Sorry, I put the comments and I put approved.

**Review by jeymarsonwolox — APPROVED** [review]
Great!!

File: src/main/java/wolox/training/controllers/BookController.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.server.ResponseStatusException;
+import wolox.training.exceptions.BookIdMismatchException;
+import wolox.training.exceptions.BookNotFoundException;
```
Comment (Jimenarosello) [inline src/main/java/wolox/training/controllers/BookController.java:17]: I think you're not using this custom exception

File: src/main/java/wolox/training/controllers/BookController.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+    @GetMapping
+    public Iterable findAll() {
+        return bookRepository.findAll();
+    }
+
+    @GetMapping("/{id}")
+    public Book findOne(@PathVariable Long id) {
+        return bookRepository.findById(id)
+                .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND,
```
Comment (Jimenarosello) [inline src/main/java/wolox/training/controllers/BookController.java]: This is where the custom exception should be 😄

**Review by Jimenarosello — CHANGES_REQUESTED** [review]
Excellent job! I have only one comment 😄

**Review by Jimenarosello — APPROVED** [review]
🚀

**Pull request merged**